Testsuite failure on 32-bits architecture

**Describe the bug**
The testsuite fails on 32-bits architecture (i386, armhf) with the following error:

```

_ TestNumpyDelta.test_numpy_delta_cases[delta_numpy7_arrays_of_different_sizes] _

self = <tests.test_delta.TestNumpyDelta object at 0xf58e0358>
t1 = array([1, 2, 3, 4]), t2 = array([ 5,  6,  7,  8,  9, 10])
deepdiff_kwargs = {}, to_delta_kwargs = {}
expected_delta_dict = {'_numpy_paths': {'root': 'int64'}, 'iterable_item_added': {'root[4]': 9, 'root[5]': 10}, 'values_changed': {'root[0]': {'new_value': 5}, 'root[1]': {'new_value': 6}, 'root[2]': {'new_value': 7}, 'root[3]': {'new_value': 8}}}
expected_result = 't2'

    @pytest.mark.parametrize(**DELTA_NUMPY_TEST_PARAMS)
    def test_numpy_delta_cases(self, t1, t2, deepdiff_kwargs, to_delta_kwargs, expected_delta_dict, expected_result):
        diff = DeepDiff(t1, t2, **deepdiff_kwargs)
        delta_dict = diff._to_delta_dict(**to_delta_kwargs)
        if expected_delta_dict:
>           assert expected_delta_dict == delta_dict
E           AssertionError: assert {'_numpy_path...w_value': 8}}} == {'_numpy_path...w_value': 8}}}
E             Omitting 2 identical items, use -vv to show
E             Differing items:
E             {'_numpy_paths': {'root': 'int64'}} != {'_numpy_paths': {'root': 'int32'}}
E             Use -v to get the full diff

tests/test_delta.py:991: AssertionError
```
Here are the complete testsuite logs, ran on the Debian package in unstable:

* i386: https://ci.debian.net/data/autopkgtest/testing/i386/d/deepdiff/20757243/log.gz
* armhf: https://ci.debian.net/data/autopkgtest/testing/armhf/d/deepdiff/20762822/log.gz

As you can see, the testsuite passes on 64-bits systems:

* amd64: https://ci.debian.net/data/autopkgtest/testing/amd64/d/deepdiff/18809180/log.gz
* arm64: https://ci.debian.net/data/autopkgtest/testing/arm64/d/deepdiff/18809250/log.gz

I am not sure if this is purely a problem with the testsuite, or if this is the result of a real bug on those systems.

**To Reproduce**
Run the testsuite on a 32-bits machines or VM via QEMU.

**Expected behavior**
The testsuite should not fail on 32-bits systems.

**OS, DeepDiff version and Python version (please complete the following information):**
 - OS: Debian
 - Version: Sid
 - Deepdiff version: 5.6.0

--- a/tests/test_delta.py
+++ b/tests/test_delta.py
@@ -2,6 +2,7 @@ import pytest
 import os
 import io
 import json
+import sys
 from decimal import Decimal
 from unittest import mock
 from deepdiff import Delta, DeepDiff
@@ -1054,7 +1055,7 @@ DELTA_NUMPY_TEST_CASES = {
                 'root[5]': 10
             },
             '_numpy_paths': {
-                'root': 'int64'
+                'root': np.where((sys.maxsize > 2**32), 'int64', 'int32')
             }
         },
         'expected_result': 't2'

